src/main/java/com/hubspot/dropwizard/guicier/GuicierApplication.java

```diff
@@ -0,0 +1,25 @@
+package com.hubspot.dropwizard.guicier;
+
+import com.google.inject.Injector;
+import io.dropwizard.Application;
+import io.dropwizard.Configuration;
+
+/**
+ * An extension of {@link Application] for application's that use a {@link GuiceBundle}.
+ *
+ * Generally speaking, you should store instance variables for the GuiceBundle and Injector.
+ * You'll want to initialize the GuiceBundle field as part of your initialize() method and
+ * you'll want to initialize the Injector field as part of your run() method. This is due to
+ * the way dropwizard initializes the Environment and Configuration objects.
+ *
+ * The injector can then be exposed via the getInjector() method you are required to implement.
+ * @param <T> the type of Configuration for the application
+ */
+public abstract class GuicierApplication<T extends Configuration> extends Application<T> {
```

Comment:
Would it be possible to drop this class and just require that the class implements `Supplier<Injector>` or something to that effect?